compiler-k2/build.gradle.kts

```
@@ -0,0 +1,35 @@
+plugins {
+  id("conventions.library")
+  id("conventions.publish")
+  kotlin("kapt")
+}
+
+conventions {
+  explicitApi = true
+}
+
+publish {
+  configurePom(
+    artifactId = "compiler-k2",
+    pomName = "Anvil Compiler Implementation for Kotlin 2+",
```

Might be worth clarifying this as the K2 compiler (or Kotlin languageLevel 2) specifically since the existing implementation also supports Kotlin 2.

```suggestion
    pomName = "Anvil Compiler Implementation for the K2 compiler",
```

(same for the name in compiler-k2-api)